patches/tModLoader/Terraria/ModLoader/PosData.cs

```diff
@@ -116,26 +130,44 @@ public static class PosData
 			return minimum;
 		}
 
-		public static bool Lookup<T>(this PosData<T>[] posMap, int x, int y, out T data) => posMap.Lookup<T>(CoordsToPos(x, y), out data);
+		public static PosData<T> Find<T>(this PosData<T>[] posMap, int pos) => posMap.FindIndex(pos) switch {
+			int i => i < 0 ? PosData<T>.nullPosData : posMap[i]
+		};
 
-		public static bool Lookup<T>(this PosData<T>[] posMap, int pos, out T data) {
-			var i = posMap.FindIndex(pos);
-			bool fail = i == -1;
-			data = fail ? default : posMap[i].value;
-			return !fail;
-		}
+		/// <summary>
+		/// General purpose lookup function. Always returns a value (even if that value is `default`).
+		/// See <see cref="PosData{T}.OrderedSparseLookupBuilder.OrderedSparseLookupBuilder(int, bool, bool)"/>for more info
+		/// </summary>
+		public static T Lookup<T>(this PosData<T>[] posMap, int x, int y) => posMap.Lookup(CoordsToPos(x, y));
```

Comment:
Modders can always directly call posData.Find() if using default values as valid values and use that to check if data is valid.